src/java.base/share/classes/java/lang/foreign/VaList.java

```diff
@@ -52,6 +52,10 @@
  * As such, this interface only supports reading {@code int}, {@code double},
  * and any other type that fits into a {@code long}.
  * <h2 id="safety">Safety considerations</h2>
+ * The behavior of any attempts to access a value in the va list, whether through one of the {@code nextVarg} overloads
+ * or {@link #skip(MemoryLayout...)}, using a memory layout other than the layout of the accessed value, or any subsequent
```

Comment:
Not super sure what you mean here by using a layout "other than the layout of the accessed value". I think you mean the layout of the underlying value that needs to be accessed? Also, is the behavior undefined even when the VaList is created safely, from Java code?

Reply:
Yes, I mean the layout of the underlying value that is being accessed. i.e. can't access a boolean as an int.

The behavior is undefined either way. Even when a VaList is created from Java, we create the same underlying native data structure (because we want to pass it to native code). This doesn't keep track of the layouts the list was built with, so there can still be mismatches.

Reply:
Right.

Reply:
This sentence still needs to be clarified IMHO. E.g. it is not clear when reading what "using a memory layout other than the layout of the accessed value" - since we are passing a layout to the access operation... it feels like we're trying to compress too much. We should say something like:

```
The values in a variable argument list are stored in one or more regions of memory in a platform specific fashion. 
Any attempt to access (or skip) any such value with a mismatched memory layout (example) will result in undefined behavior.
```

Reply:
I think we don't need to say anything about how the values are stored. I wanted to avoid using the term "undefined behavior" at first as well, since it sounds like something that is defined elsewhere (which it isn't). Though, I agree that what I have now doesn't quite feel right.

Reply:
I've tried to re-write this to be more clear (including an example)